src/Universe.cpp

```diff
@@ -1709,8 +1711,8 @@ Cell* Lattice::findCell(LocalCoords* coords) {
   double next_z = coords->getZ()
                   - (getMinZ() + _widths_z[lat_z]/2. + _accumulate_z[lat_z]);
 
-  /* Check for 2D problem */
-  if (_width_z == std::numeric_limits<double>::infinity())
+  /* Check for 2D problem or 2D lattice */
+  if (_width_z > FLT_INFINITY)
```

Comment:
I ran into the case that the line above was not working. I think it's very much bad luck since the double is supposed to be set to this value exactly. 
Either way, I think this is a good substitute and we should think about using it in more places.

Reply:
Good point